### PR TITLE
Application Server 6.0.0 improvements

### PR DESCRIPTION
**Pull request by Mirage20**
* used appserver-util api to retrive the configurations to webapp-loader module
* added custom exceptions to the webapp-loader module and appserver-util modules
* remove unwanted classes in webapp-loader and appserver-utill modules
* webapp-loader performance improvements in debug logging
* improved exception handling
* removed unwanted constants in webapp-loader and appserver-utill modules
* extracted JAXB inner bean classes as normal bean classes
* improved log messages.
* minor code imprivements in webapp-loader and appserver-utill modules

### CODE REVIEW

**Comment by Nipuni** [comment]
Please resolve conflicts and update the PR

File: modules/appserver-utils/src/main/java/org/wso2/appserver/Constants.java [hunk 1]
```diff
@@ -37,6 +37,8 @@
 
     public static final String WEBAPP_RESOURCE_FOLDER = "WEB-INF";
 
+
```
Comment (manoj-kumara) [inline modules/appserver-utils/src/main/java/org/wso2/appserver/Constants.java]: Remove unwanted comments.

File: modules/appserver-utils/src/main/java/org/wso2/appserver/configuration/context/ContextConfiguration.java [hunk 2]
```diff
@@ -51,6 +51,7 @@ public void setSingleSignOnConfiguration(SSOConfiguration singleSignOnConfigurat
         this.singleSignOnConfiguration = singleSignOnConfiguration;
     }
 
+
```
Comment (manoj-kumara) [inline modules/appserver-utils/src/main/java/org/wso2/appserver/configuration/context/ContextConfiguration.java]: Remove unwanted comments.

File: modules/appserver-utils/src/main/java/org/wso2/appserver/exceptions/ConfigurationException.java [hunk 3]
```diff
@@ -14,21 +14,23 @@
...
+ * Thrown when the application server failed to build the configurations.
+ *
+ * @since 6.0.0
  */
-public class ApplicationServerException extends Exception {
+public class ConfigurationException extends ApplicationServerException {
+    private static final long serialVersionUID = -3620848532234792303L;
 
-    public ApplicationServerException(String message, Throwable cause) {
+    public ConfigurationException(String message, Throwable cause) {
```
Comment (manoj-kumara) [inline modules/appserver-utils/src/main/java/org/wso2/appserver/exceptions/ConfigurationException.java]: I think last we discussed to chenge this to ApplicationServerConfigurationException

File: modules/appserver-utils/src/main/java/org/wso2/appserver/configuration/context/ContextConfiguration.java (hunk 2 above)
Comment (Mirage20) [inline modules/appserver-utils/src/main/java/org/wso2/appserver/configuration/context/ContextConfiguration.java]: which comments?

File: distribution/contents/conf/wso2/wso2as-web.xml [hunk 4]
```diff
@@ -12,10 +12,10 @@
  limitations under the License.
 -->
 <wso2as-web xmlns="http://wso2.org/2016/wso2as-web">
-    <classloading>
-        <parent-first>true</parent-first>
+    <class-loader>
+        <parent-first>false</parent-first>
         <environments>CXF</environments>
```
Comment (manoj-kumara) [inline distribution/contents/conf/wso2/wso2as-web.xml]: This should be tomcat by default ryt ?

File: modules/webapp-loader/src/main/java/org/wso2/appserver/webapp/loader/AppServerWebappClassLoader.java [hunk 5]
```diff
@@ -23,83 +23,83 @@
...
         if (clazz != null) {
-            log.debug("  Returning class from local cache (WebappClassLoader)");
+            if (log.isDebugEnabled()) {
+                log.debug("  Returning class from local cache (WebappClassLoader)");
+            }
             if (resolve) {
                 resolveClass(clazz);
             }
             return (clazz);
         }
```
Comment (manoj-kumara) [inline modules/webapp-loader/src/main/java/org/wso2/appserver/webapp/loader/AppServerWebappClassLoader.java]: IIRC we discussed to remove this one and test Tomcat context configuration for reading the parent first behaviour. Let me know you did that ?

File: distribution/contents/conf/wso2/wso2as-web.xml (hunk 4 above)
Comment (Mirage20) [inline distribution/contents/conf/wso2/wso2as-web.xml]: We don't need to add tomcat it self as a environment. I will remove this in next PR

File: modules/webapp-loader/src/main/java/org/wso2/appserver/webapp/loader/AppServerWebappClassLoader.java (hunk 5 above)
Comment (Mirage20) [inline modules/webapp-loader/src/main/java/org/wso2/appserver/webapp/loader/AppServerWebappClassLoader.java]: Currently I'm working on it. If it changes the util module and xsd's will also need to modify

**Pull request merged**